### PR TITLE
App and user for tracing

### PR DESCRIPTION
**Pull request by t-ronmoneta**
### Added
app and user for tracing fields in ingestion properties
Took those fields from KCSB and added them to ingestion Propeties in queued ingest client

### CODE REVIEW

**Review by ohadbitt — APPROVED** [review]
Just make sure it works :)

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java [hunk 1]
```diff
@@ -57,6 +59,8 @@ public class QueuedIngestClientImpl extends IngestClientBase implements QueuedIn
         this.resourceManager = new ResourceManager(client, httpClient);
         this.azureStorageClient = new AzureStorageClient();
         this.connectionDataSource = csbWithEndpoint.getClusterUrl();
+        this.applicationForTracing = csb.getApplicationNameForTracing();
```
Comment (AsafMah) [inline ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]: Create a client details object and use that - these properties are not formatted/have the wrong defaults

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionBlobInfo.java [hunk 2]
```diff
@@ -130,4 +134,20 @@ public ValidationPolicy getValidationPolicy() {
     public void setValidationPolicy(ValidationPolicy validationPolicy) {
         this.validationPolicy = validationPolicy;
     }
+
+    public String getApplicationForTracing() {
```
[the remark this answers is not in the document]
Comment (t-ronmoneta) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionBlobInfo.java:138]: Common practice, can (and will) be removed

File: ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java (hunk 1 above)
Comment (t-ronmoneta) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]: When looking at the client details object current usage - the appendedClientVersionForTracing is either null, or csb.getApplicationNameForTracing() as done here. so not sure why this is wrong

Comment (AsafMah) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]:
```
 constructor(public applicationNameForTracing: string | null, public userNameForTracing: string | null) {
        if (this.applicationNameForTracing === null) {
            this.applicationNameForTracing = ClientDetails.defaultApplication();
        }
        if (this.userNameForTracing === null) {
            this.userNameForTracing = ClientDetails.defaultUser();
        }
        this.versionForTracing = ClientDetails.defaultVersion();
    }
````

Comment (t-ronmoneta) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]: image
this is the usage so far as well in the client details too though - so using clientdetails would give you the exact same thing

Comment (AsafMah) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]: The getter in ClientDetails returns the default if it's null - that's the piece of code I showed you.
I'm not sure what do you mean

Comment (t-ronmoneta) [reply ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClientImpl.java:62]: ohhh in applilcation tracing, all the time i looked at the second one, which is indded the same. ok, changing

**Review by AsafMah — APPROVED** [review]

**Pull request merged**